### ISSUE TITLE
task dag enhancement.

### ISSUE THREAD

**Issue by kyungjoo-kim** [kokkos/kokkos#1354]
There are missing features in Kokkos task dag capability regarding to CUDA space
- user interface to set cuda stack size;
- per team and per thread;
- user interface to set vector size currently 1x32 is hardwired. 

I sent a patch to @hcedwar and this issue is for a tracking purpose.

**Comment by hcedwar** [comment]
@kyungjoo-kim : Integraton testing fixes to CUDA back-end

The code is finally working. I need to have kokkos to be updated with the attached patch.
After we fixed capturing the member variables in a class, the error related with "std::move" does not appear anymore and I removed the modification.
The patch includes.
- memory pool kokkos inline decoration
- task team member need to have team_barrier() and team_broadcast.
- I reduced the stack size to default 1024. I tested it and it does not need larger stack anymore (but who knows someday I need it again; so I do not completely remove the interface in the kokkos.
